### DISCUSSION TITLE
Newbe: stream i2s to i2s and more conceptional questions?

### DISCUSSION THREAD

**Discussion by ITServ-DE** [pschatzmann/arduino-audio-tools#1366] [Q&A] [answered]
Hi Community,
I'm new to Aduino on ESP32-WROOM-32, and (of course) I have to start with something complicated. For that, I'd like to check first if it's even possible what I want to do.

Is it possible 
1. to stream from an i2s input to an i2s output (MEMS microphone to PCM5102 module)? ESP32 seems to have 2 I2S peripherals ...
2. use an external 16 bit ADC connected via I²C instead of the internal 12 bit ADC to sample audio with an analog source?
3. directly connect i2s to i2s even without a microcontroller?
4. Where can I find how to/which commands are used to set the pins for an i2s connection and override the default pins (required ?

With this question, I evaluate my options, so thank you very much for pushing me into the right direction.

**Comment by pschatzmann** [comment, marked as answer]
First a small correction: it's i2s and not a2s!
1) yes - i2s can read and write at the same time on one port or you can use the two ports. There are separate examples for each design.
2) regular i2c ADCs are too slow for audio but you can use the internal ADC. Note it needs to be on the I2S port 0!
3) never tried: but one needs to be a master and the other a client: so you will need the possibility to configure at least one of the devices
4) in the Wiki External DACs

**Reply by ITServ-DE** [reply]
Thank you for your reply, so I know in which direction I will work and learn.

Let me ask one more thing, please. Espressif provides an "Audio Front-End" which has noise suppression (Rauschunterdrückung) and echo cancellation. I did not see such filters in your code. Given that I'm new to Arduino/C++, as I said, do you have such filters in your code?

**Comment by pschatzmann** [comment]
No, I don't. 
If you want to work on this: Any contribution is welcome....

**Reply by ITServ-DE** [reply]
OK, thank you very much.
